### ISSUE TITLE
Unable to disable transactions on a per migration basis when using knex migrate:up 

### ISSUE THREAD

**Issue by andykapteyn** [knex/knex#4539]
# Environment

Knex version: 0.95.6
Database + version: PostgreSQL 10.15
OS: WSL: Ubuntu-18.04 

# Bug

1. Explain what kind of behaviour you are getting and how you think it should do

I'm trying to run a migration using `knex migrate:up` with transactions disabled by including `export const config = { transaction: false };` (as documented here: http://knexjs.org/#Migrations-API-transactions) but the migration is still being run within a transaction. When the same migration is run with `knex migrate:latest` the migration is successfully run with transactions disabled.

Looking at the code I believe this is due to the following line returning a promise which hasn't been awaited resulting in `this._useTransaction` defaulting to the global setting.
https://github.com/knex/knex/blob/22002fd8de440fe5e94ce189307d5aa5d6058f41/lib/migrations/migrate/Migrator.js#L145
![image](https://user-images.githubusercontent.com/23250032/123582068-40609280-d831-11eb-8cbd-e55acb4e4634.png)

The `knex migrate:latest` functionality does await the result of `this.config.migrationSource.getMigration(...);` which explains the difference in functionality.
https://github.com/knex/knex/blob/22002fd8de440fe5e94ce189307d5aa5d6058f41/lib/migrations/migrate/Migrator.js#L86

2. Error message

N/A as any errors that are displayed are specific to the migrations being run.

3. Reduced test code, for example in https://npm.runkit.com/knex or if it needs real
   database connection to MySQL or PostgreSQL, then single file example which initializes
   needed data and demonstrates the problem.

Initalise the migration project using the default settings (the example uses TypeScript) and set up a PostgreSQL connection.

Create a single migration:
```
import { Knex } from 'knex';

export async function up(knex: Knex): Promise<void> {
    const isTransaction = await knex.raw('SELECT now() <> statement_timestamp() AS "is_transaction";')
        .then(response => response.rows[0]['is_transaction']);

    console.log(isTransaction);
}

export async function down(_: Knex): Promise<void> {
}

export const config = { transaction: false };
```

If the migration is run using `knex migrate:up` the result will be `true`. If it's run with `knex migrate:latest` the result will be `false`.

Let me know if you need any further information.

**Comment by kibertoad** [comment]
@andykapteyn Thank you for the report! Would you be open to submit a PR (including test) for this issue?

**Comment by HummingBird24** [comment]
+1

**Comment by andykapteyn** [comment]
@HummingBird24 I believe the fix for this was included in [v0.95.11](https://github.com/knex/knex/releases/tag/0.95.11) (see #4550).

**Comment by HummingBird24** [comment]
@andykapteyn you're right. I didn't know that multiple statements in knex.schema.raw automatically created a transaction block. After separating them, it worked. Thanks for the quick response!
